### ISSUE TITLE
hyperkit port forwarding not working on MacOS Catalina

### ISSUE THREAD

**Issue by nustiueudinastea** [linuxkit/linuxkit#3497]
I have been trying to run a linuxkit VM on MacOS Catalina and forward port 2222 as explained in the [hyperkit doc file](https://github.com/linuxkit/linuxkit/blob/master/docs/platform-hyperkit.md).

I have tried both networking modes **docker-for-mac** and **vpnkit** with similar results, but this bug report is for the **docker-for-mac** mode where vpnkit is shared. I used the [sshd example](https://github.com/linuxkit/linuxkit/blob/master/examples/sshd.yml) and built a vm that I ran using **linuxkit hyperkit**. 

**Software versions**

hyperkit: v0.20200224-18-gf1dd3c
docker desktop for mac: 2.2.3.0 (43965), edge channel

**Steps to reproduce the issue:**

1. Have docker running
2. Run linuxkit using:
```
$ linuxkit run hyperkit -networking docker-for-mac -publish 2222:22/tcp -iso -uefi vpnkit-efi
```

**Describe the results you received:**

```
$ linuxkit run hyperkit -networking docker-for-mac -publish 2222:22/tcp -iso -uefi vpnkit-efi
2020/04/14 19:43:49 Dialling unix /Users/user/Library/Containers/com.docker.docker/Data/vpnkit.port.sock
2020/04/14 19:44:19 Failed to establish 9P session to read unix ->vpnkit.port.sock: i/o timeout
FATA[0030] Publish ports failed with: Connection to VPNKit failed: read unix ->vpnkit.port.sock: i/o timeout
```

**Describe the results you expected:**

I expected the VM to run successfully, after which to be able to SSH from the macOS host to the linuxkit VM, using port 2222.

**Comment by justincormack** [comment]
cc @djs55  has anything changed here? i/o error is weird.

**Comment by djs55** [comment]
I think @guillaumerose probably knows the current state of the vpnkit port forwarding better than me. Possibly something needs revendoring?

**Comment by guillaumerose** [comment]
Yes, I updated the vendoring few weeks ago. See https://github.com/linuxkit/linuxkit/pull/3479

@nustiueudinastea Which version of linuxkit are you using ? Can you try with latest master ?

**Comment by nustiueudinastea** [comment]
@guillaumerose I am using the following version:

```
$ linuxkit version
linuxkit version v0.7+
commit: afff608c499b27a9b90b01dc173857a93f914524
```

I will try master some time today and report back.
